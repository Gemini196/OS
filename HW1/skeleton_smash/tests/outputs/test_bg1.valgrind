==44820== Memcheck, a memory error detector
==44820== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==44820== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==44820== Command: /home/student/Desktop/OS/HW1/skeleton_smash/smash
==44820== Parent PID: 44816
==44820== 
==44820== 
==44820== FILE DESCRIPTORS: 6 open at exit.
==44820== Open file descriptor 1: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_bg1.out
==44820==    at 0x54EF957: dup2 (syscall-template.S:78)
==44820==    by 0x112CC9: main (smash.cpp:41)
==44820== 
==44820== Open file descriptor 0:
==44820==    at 0x54EF957: dup2 (syscall-template.S:78)
==44820==    by 0x112C77: main (smash.cpp:33)
==44820== 
==44820== Open file descriptor 5:
==44820==    at 0x54EF927: dup (syscall-template.S:78)
==44820==    by 0x112E17: __static_initialization_and_destruction_0(int, int) (smash.cpp:9)
==44820==    by 0x112E33: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==44820==    by 0x112EAC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==44820==    by 0x5400C17: (below main) (libc-start.c:266)
==44820== 
==44820== Open file descriptor 4: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_bg1.out
==44820==    at 0x54EF927: dup (syscall-template.S:78)
==44820==    by 0x112E07: __static_initialization_and_destruction_0(int, int) (smash.cpp:8)
==44820==    by 0x112E33: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==44820==    by 0x112EAC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==44820==    by 0x5400C17: (below main) (libc-start.c:266)
==44820== 
==44820== Open file descriptor 3: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_bg1.valgrind
==44820==    <inherited from parent>
==44820== 
==44820== Open file descriptor 2: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_bg1.err
==44820==    <inherited from parent>
==44820== 
==44820== 
==44820== HEAP SUMMARY:
==44820==     in use at exit: 163 bytes in 2 blocks
==44820==   total heap usage: 21 allocs, 19 frees, 82,277 bytes allocated
==44820== 
==44820== 3 bytes in 1 blocks are indirectly lost in loss record 1 of 2
==44820==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==44820==    by 0x10B5B9: _parseCommandLine(char const*, char**) (Commands.cpp:47)
==44820==    by 0x10D7C9: BackgroundCommand::execute() (Commands.cpp:564)
==44820==    by 0x10F690: SmallShell::executeCommand(char const*) (Commands.cpp:971)
==44820==    by 0x112D72: main (smash.cpp:51)
==44820== 
==44820== 163 (160 direct, 3 indirect) bytes in 1 blocks are definitely lost in loss record 2 of 2
==44820==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==44820==    by 0x10D781: BackgroundCommand::execute() (Commands.cpp:558)
==44820==    by 0x10F690: SmallShell::executeCommand(char const*) (Commands.cpp:971)
==44820==    by 0x112D72: main (smash.cpp:51)
==44820== 
==44820== LEAK SUMMARY:
==44820==    definitely lost: 160 bytes in 1 blocks
==44820==    indirectly lost: 3 bytes in 1 blocks
==44820==      possibly lost: 0 bytes in 0 blocks
==44820==    still reachable: 0 bytes in 0 blocks
==44820==         suppressed: 0 bytes in 0 blocks
==44820== 
==44820== For counts of detected and suppressed errors, rerun with: -v
==44820== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
