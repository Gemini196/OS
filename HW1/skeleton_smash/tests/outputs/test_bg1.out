smash> smash: got ctrl-Z
smash: process 49812 was stopped
smash> smash> 